### DISCUSSION TITLE
Searching in pdf via 'find' dispatch event always results in state as Pending - pdfjs-dist-4.6.82

### DISCUSSION THREAD

**Discussion by amitduaaa** [mozilla/pdf.js#18872] [Q&A]
I am using pdfjs-dist-4.6.82 on a react project and I have implemented a PDF viewer. I am unable to add a find functionality in the pdfjs-dist-4.6.82 version.

Here how my code for find functionality looks like: 

a method that dispatches the 'find' event, handleNextsearch that has type as again, two event listeners.
```
	useEffect(() => {
		const initializeController = () => {
			pdfjsViewer.eventBus.on('updatefindcontrolstate', handleUpdateState);
			pdfjsViewer.eventBus.on('updatefindmatchescount', handleUpdateResultsCount);
		};

		initializeController();

		return () => {
			pdfjsViewer.eventBus.off('updatefindcontrolstate', handleUpdateState);
			pdfjsViewer.eventBus.off('updatefindmatchescount', handleUpdateResultsCount);
		};
	}, [pdfjsViewer]);

	const handleSearch = (query: string) => {
		if (query === currentSearch) {
			handleNextSearchItem();
			return;
		}

		executeCommand('find', { type: 'highlightallchange', query });
		setCurrentSearch(query);
		setSearchIndex(0);
	};

	const handleNextSearchItem = () => {
		executeCommand('find', { type: 'again', query: currentSearch });
		setSearchIndex(prevIndex => Math.min((prevIndex || 0) + 1, (itemsFound || 0) - 1));
	};

	const handlePrevSearchItem = () => {
		executeCommand('find', { findPrevious: true, query: currentSearch });
		setSearchIndex(prevIndex => Math.max((prevIndex || 0) - 1, 0));
	};

	const executeCommand = async (command: any, options: FindControllerCommand = {}) => {
		await pdfjsViewer?.findController!._eventBus.dispatch(command, {
			...baseFindOptions,
			...options,
		});
	};
```
If I search for a word and hit enter, I see all the results highlighted but the updatefindcontrolstate event never triggers. If I now try to search for next result, updatefindcontrolstate event triggers but the state is stuck at pending i.e. 3. I also get the following error in console: There can only be one pending page.

If I now search for new word and hit enter, it would highlight all the results but then we're back to the same above issue. It's also strange that first result doesn't have green highlight on its own which is something that used to happen in older versions 2.x.x. 

I couldn't figure out why it's behaving like this and hence reaching out to community for help and support. 

Let me know if you need any other information from my side.

**Comment by nicolo-ribaudo** [comment]
Could you create a reproduction without React?
